src/transformStateWithClones.js

```diff
@@ -6,8 +6,47 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const newState = { ...state };
+  const transformLogState = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        addProperties(newState, action.extraData);
+        transformLogState.push({ ...newState });
+        break;
+
+      case 'removeProperties':
+        removeProperties(newState, action.keysToRemove);
+        transformLogState.push({ ...newState });
+        break;
+
+      case 'clear':
+        clearProperties(newState);
+        transformLogState.push({ ...newState });
+        break;
+    }
```

Comment:
[CODE STYLE]: Use a `default` case in your `switch` statement for error handling. This will ensure that if an action with an unrecognized type is passed, the function can handle it appropriately, such as throwing an error or ignoring the action.